### PR TITLE
[BugFix] Fix jit constant error

### PR DESCRIPTION
**Pull request by Moonm3n**
Why I'm doing:

When jit is enabled, there will be an error when running this sql:
`select bitnot(cast(power(-2,127) as largeint)+1);`

What I'm doing:

Fix crash error caused by jit compilation.

Fixes https://github.com/StarRocks/StarRocksTest/issues/5556

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exprs/jit/jit_expr.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ StatusOr<ColumnPtr> JITExpr::evaluate_checked(starrocks::ExprContext* context, C
     }
 #endif
 
-    auto result_column = ColumnHelper::create_column(type(), is_nullable(), is_constant(), ptr->num_rows(), false);
+    auto result_column = ColumnHelper::create_column(type(), !is_constant() && is_nullable(), is_constant(), ptr->num_rows(), false);
```
Comment (fzhedu) [inline be/src/exprs/jit/jit_expr.cpp:111]: remove `unpack_and_duplicate_const_column` in Line:98

**Review by fzhedu — COMMENTED** [review]
add the test case

**Review by fzhedu — APPROVED** [review]

File: be/src/exprs/jit/jit_functions.cpp [hunk 2]
```diff
@@ -258,16 +259,16 @@ Status JITFunction::llvm_function(JITScalarFunction jit_function, const Columns&
...
-        auto data_column = FunctionHelper::get_data_column_of_nullable(column);
+        auto data_column = ColumnHelper::get_data_column(column.get());
         auto datums = reinterpret_cast<const int8_t*>(data_column->raw_data());
 
         const int8_t* null_flags = nullptr;
-        if (column->is_nullable()) {
+        if (!column->is_constant() && column->is_nullable()) {
             null_flags = reinterpret_cast<const int8_t*>(
                     ColumnHelper::as_raw_column<NullableColumn>(column)->null_column()->raw_data());
         }
```
Comment (satanson) [inline be/src/exprs/jit/jit_functions.cpp:269]: there are 4 types of columns:
1. constant column; 
2. constant nullable column;
3. nullable column;
4. column;
it seems only type-4 and type-3 are considerated, what if type-1 and type-2 column?

Comment (fzhedu) [reply be/src/exprs/jit/jit_functions.cpp:269]: here only get null map for nullable columns

File: be/src/exprs/jit/jit_expr.cpp (hunk 1 above)
Comment (Moonm3n) [reply be/src/exprs/jit/jit_expr.cpp]: Done.

**Review by Seaven — APPROVED** [review]

**Pull request merged**